### PR TITLE
tracy: init at 0.6.3

### PR DESCRIPTION
**Pull request by mpickering**
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ethercrow** [comment]
What about `import-chrome` executable?

File: pkgs/development/tools/tracy/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, tbb, gtk2, glfw, pkgconfig}:
+stdenv.mkDerivation rec {
+  name    = "tracy-${version}";
+  version = "0.6.3";
+
+  buildInputs = [tbb gtk2 glfw pkgconfig];
```
Comment (maksbotan) [inline pkgs/development/tools/tracy/default.nix:6]: Why is `freetype` missing from here?
Also, I suppose that `pkgconfig` should go to `nativeBuildInputs`.

File: pkgs/development/tools/tracy/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildPhase = ''
+    make -C profiler/build/unix/ CFLAGS=-ltbb release
```
Comment (maksbotan) [inline pkgs/development/tools/tracy/default.nix:9]: On macOS I get this error

```
In file included from ../../../server/TracyThreadCompress.cpp:3:
../../../server/TracyFileRead.hpp:42:20: error: aligned allocation function of type 'void *(unsigned long, enum std::align_val_t)' is only available on macOS 10.13 or newer
        return f ? new FileRead( f, fn ) : nullptr;
```

You need to add `-faligned-allocation` to `CFLAGS` (on the next line as well) to fix it.

See also #73744.

File: pkgs/development/tools/tracy/default.nix (hunk 1 above)
Comment (maksbotan) [reply pkgs/development/tools/tracy/default.nix:6]: Also, for macos you need to add `Carbon AppKit` from `pkgs.darwing.apple_sdk.frameworks`.
I've seen somewhere snippets for conditional adding of these deps...

`gtk`, on the other hand, is not needed on macOS.

File: pkgs/development/tools/tracy/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./profiler/build/unix/Tracy-release $out/bin/Tracy
+    cp ./import-chrome/build/unix/import-chrome-release $out/bin/import-chrome
+  '';
+
+
+  src = fetchFromGitHub {
+    owner    = "wolfpld";
```
Comment (maksbotan) [inline pkgs/development/tools/tracy/default.nix:21]: Strange looking spaces :)

**Comment by mpickering** [comment]
@GrahamcOfBorg build tracy

File: pkgs/development/tools/tracy/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+
+  src = fetchFromGitHub {
+    owner = "wolfpld";
+    repo = "tracy";
+    rev = "v${version}";
+    sha256 = "0pgq8h5gq141zq1k4cgj6cp74kh4zqbp7h4wh29q4grjb04yy06i";
+  };
+
+
```
Comment (Mic92) [inline pkgs/development/tools/tracy/default.nix:34]:
```suggestion

```

File: pkgs/development/tools/tracy/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    make -C import-chrome/build/unix/ CFLAGS="$CFLAGS" release
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./profiler/build/unix/Tracy-release $out/bin/Tracy
+    cp ./import-chrome/build/unix/import-chrome-release $out/bin/import-chrome
+  '';
+
+
```
Comment (Mic92) [inline pkgs/development/tools/tracy/default.nix:26]:
```suggestion

```

File: pkgs/development/tools/tracy/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, darwin, fetchFromGitHub, tbb, gtk2, glfw, pkgconfig, freetype}:
+stdenv.mkDerivation rec {
+  name    = "tracy-${version}";
+  version = "0.6.3";
+
+  buildInputs =
+    [glfw pkgconfig]
```
Comment (Mic92) [inline pkgs/development/tools/tracy/default.nix:7]: pkgconfig needs to be in nativeBuildInputs.

File: pkgs/development/tools/tracy/default.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+             + lib.optionalString stdenv.cc.isClang "-faligned-allocation";
+  buildPhase = ''
+    make -C profiler/build/unix/ CFLAGS="$CFLAGS" release
+    make -C import-chrome/build/unix/ CFLAGS="$CFLAGS" release
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./profiler/build/unix/Tracy-release $out/bin/Tracy
+    cp ./import-chrome/build/unix/import-chrome-release $out/bin/import-chrome
```
Comment (Mic92) [inline pkgs/development/tools/tracy/default.nix:23]:
```suggestion
    install -D ./profiler/build/unix/Tracy-release $out/bin/Tracy
    install -D ./import-chrome/build/unix/import-chrome-release $out/bin/import-chrome
```

File: pkgs/development/tools/tracy/default.nix (hunk 6 above)
Comment (mpickering) [reply pkgs/development/tools/tracy/default.nix:7]: Why is this?

Comment (mpickering) [reply pkgs/development/tools/tracy/default.nix:7]: @Mic92 See https://github.com/NixOS/nixpkgs/issues/89389

**Comment by mpickering** [comment]
All review comments resolved.

File: pkgs/development/tools/tracy/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs =
+    [glfw]
+    ++ lib.optionals stdenv.isDarwin [darwin.apple_sdk.frameworks.Carbon
+                                      darwin.apple_sdk.frameworks.AppKit
+                                      freetype ]
+    ++ lib.optionals stdenv.isLinux [gtk2 tbb];
+
+  CFLAGS = lib.optionalString stdenv.isLinux "-ltbb"
+             + lib.optionalString stdenv.cc.isClang "-faligned-allocation";
```
Comment (Mic92) [inline pkgs/development/tools/tracy/default.nix:15]: This will be invalid if both isLinux and isClang is true because of a missing space.

**Comment by Mic92** [comment]
@GrahamcOfBorg build tracy

**Comment by mpickering** [comment]
Thanks for pushing your fixes @Mic92

**Pull request merged**